### PR TITLE
Add Both mode for Supported LitShaderMode

### PR DESCRIPTION
**Pull request by RSlysz**
### Purpose of this PR
Add Both mode for supported rendering path in HDRenderPipelineAsset

---
### Release Note
Multiple Rendering Path supported in parallel. In mode Both, Each renderer can be either Forward or Deferred. 
HDRenderPipelineAsset's cache size are now delayed fields.

---
### Testing status
**Katana Tests**: 

**Manual Tests**: Tested locally

**Automated Tests**: 

---
### Overall Product Risks
**Technical Risk**: Hight
(Numerous core files modified)

**Halo Effect**: Hight
(Several core rendering feature impacted)

---

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.cs [hunk 1]
```diff
@@ -10,12 +10,11 @@ partial class FrameSettingsUI : BaseUI<SerializedFrameSettings>
         public AnimBool isSectionExpandedRenderingSettings { get { return m_AnimBools[2]; } }
         public AnimBool isSectionExpandedXRSettings { get { return m_AnimBools[3]; } }
         public AnimBool isSectionExpandedXRSupported { get { return m_AnimBools[4]; } }
-        public AnimBool isSectionExpandedUseForwardOnly { get { return m_AnimBools[5]; } }
 
         public LightLoopSettingsUI lightLoopSettings = new LightLoopSettingsUI();
 
         public FrameSettingsUI()
-            : base(7)
+            : base(6)
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/FrameSettingsUI.cs]: We might want a named constant here or at least a comment.

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/RenderPipelineSettingsUI.cs [hunk 2]
```diff
@@ -90,14 +90,22 @@ static void Drawer_SectionPrimarySettings(RenderPipelineSettingsUI s, Serialized
...
             // MSAA is an option that is only available in full forward but Camera can be set in Full Forward only. Thus MSAA have no dependency currently
-            EditorGUILayout.PropertyField(d.supportMSAA, _.GetContent("Support Multi Sampling Anti-Aliasing|This feature doesn't work currently."));
-            using (new EditorGUI.DisabledScope(!d.supportMSAA.boolValue))
+            //Note: do not use SerializedProperty.enumValueIndex here as this enum not start at 0 as it is used as flags.
+            bool isForwardOnly = d.supportedRenderingPath.intValue == (int)UnityEngine.Experimental.Rendering.HDPipeline.RenderPipelineSettings.SupportedRenderingPath.ForwardOnly;
+            using (new EditorGUI.DisabledScope(!isForwardOnly))
             {
                 ++EditorGUI.indentLevel;
-                EditorGUILayout.PropertyField(d.MSAASampleCount, _.GetContent("MSAA Sample Count|Allow to select the level of MSAA."));
+                d.supportMSAA.boolValue = EditorGUILayout.Toggle(_.GetContent("Support Multi Sampling Anti-Aliasing|This feature doesn't work currently."), d.supportMSAA.boolValue && isForwardOnly);
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/Settings/RenderPipelineSettingsUI.cs]: "This feature doesn't work currently."
Isn't MSAA officially supported now? Shouldn't we remove that?

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/FrameSettings.cs [hunk 3]
```diff
@@ -97,7 +103,7 @@ public class FrameSettings
         public float specularGlobalDimmer = 1.0f;
 
         // View
-        public bool enableForwardRenderingOnly = false; // TODO: Currently there is no way to strip the extra forward shaders generated by the shaders compiler, so we can switch dynamically.
+        public LitShaderMode shaderLitMode = LitShaderMode.Deferred; // TODO: Currently there is no way to strip the extra forward shaders generated by the shaders compiler, so we can switch dynamically.
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Settings/FrameSettings.cs]: We can remove this comment now I think.

**Comment by sebastienlagarde** [comment]
Due to new SRP API change on master we have katana broken here, so it require a new master merge. However we need this PR to fix MSAA, so I a merging this PR that have been tested locally so MSAA can be fixed in time.

**Pull request merged**